Add docstrings

caddycarine opened the pull request:
Add docstrings to strategy methods so they don't display the default placeholder docstring and simplify chained comparisons in SelfSteem class

czemtsop committed:
Ran black on all files after cherry-pick

czemtsop committed:
Cherry-pick merge

czemtsop committed:
Redoing unmerged changes:

Added docstrings to files which had not merged.
Simplified chained comparisons in SelfSteem
Changed set(["game"]) to {"game"} to reduce function call
Deleted duplicate TestIdentityDualTransformer class

czemtsop committed:
Resolving merge conflicts

czemtsop committed:
Resolving merge conflicts

drvinceknight commented on .isort.cfg:
```diff
@@ -4,4 +4,4 @@ multi_line_output = 3
 force_grid_wrap = 0
 combine_as_imports = True
-line_length = 80
+line_length = 88
```
This line here is causing a number of other things to break.

caddycarine replied:
`python -m isort --recursive axelrod/.` and `python -m black -l 80` .
seem to do conflicting changes.

drvinceknight replied:
They shouldn't based on the setting in https://github.com/Axelrod-Python/Axelrod/blob/dev/.isort.cfg but it could be that the specific version of `isort` you're using isn't the one we have setup in the CI: https://github.com/Axelrod-Python/Axelrod/blob/dev/.github/workflows/config.yml#L52

(We should clarify that somewhere.)

drvinceknight replied:
Sorry this PR is proving to be a pain for you, if there's something specific I can do to assist please ask. We'll have to have a think afterwards how we can avoid the friction you're experiencing. It hasn't really happened in the past but we've made  a number of small tweaks to our CI recently which might be the cause.

caddycarine replied:
![isort](https://user-images.githubusercontent.com/7005763/95219793-52069380-07e5-11eb-916f-eb38fcf93859.png)
I  can't get both of them to agree on these four files:
1. axelrod\strategies\titfortat.py
2. axelrod\tests\property.py
3. axelrod\tests\strategies\test_evolvable_player.py
4. axelrod\tests\strategies\test_finite_state_machines.py

Would appreciate any suggestions/recommendations.

drvinceknight replied:
Taking a look now :)

drvinceknight replied:
I've opened https://github.com/caddycarine/Axelrod/pull/1 on your fork. This is a PR to this branch so once you merge that it should update this PR and __should__ get the CI to pass.

The main problem seemed to be https://github.com/caddycarine/Axelrod/pull/1/commits/bbe9a7238f75f7362425c2a43aede6f0bbafd87c

caddycarine replied:
Your solution works 😁!
Thanks.

czemtsop committed:
Resolve merge issues

Modified .isort.cfg to resolve merge issues.

czemtsop committed:
Resolve merge issues

1. Modified max.size in strategy_lists because len(axl.short_run_time_strategies) < len(axl.strategies) in property.py

2. Added classifiers_list method, in test_filtering.py, and used it in test_boolean_filtering.

drvinceknight committed:
Fix isort configuration file.

drvinceknight committed:
Run black and isort.

caddycarine committed:
Merge pull request #1 from Axelrod-Python/caddycarine-add_docstrings

Fix isort config

drvinceknight commented on axelrod/tests/unit/test_strategy_transformers.py:
```diff
@@ -1001,20 +1001,6 @@ class TestFlippedDualTransformer(TestPlayer):
     }
 
 
-class TestIdentityDualTransformer(TestPlayer):
```
Why remove this?

caddycarine replied:
Repetition; an identical class was defined earlier..

drvinceknight replied:
Great :+1: Thank you.

drvinceknight reviewed (approved):
This looks good to me. :+1:
